quadsim/robots/motors.py

```diff
@@ -1,48 +1,54 @@
-""" Implements models for single DC-DC motors and compositional motor groups.
-"""
+"""Implements models for single DC motors and compositional motor groups."""
 import enum
+from dataclasses import dataclass
 from typing import Optional
 from typing import Sequence
 from typing import Tuple
 from typing import Union
 
 import numpy as np
 
-POSITION_INDEX = 0
-KP_INDEX = 1
-VELOCITY_INDEX = 2
-KD_INDEX = 3
-TORQUE_INDEX = 4
-COMMAND_DIMENSION = 5
+_ARRAY = Sequence[float]
+_FloatOrArray = Union[float, _ARRAY]
 
 
 class MotorControlMode(enum.Enum):
     """Different motor control modes.
-    See the documentation of MotorModel class for details."""
+
+    Right now, 3 motor control modes are supported:
+    - POSITION: performs joint-level PD control.
+    - TORQUE: directly takes in motor torque command
+    - HYBRID: takes in a 5-d tuple (pos, kp, vel, kd, torque), and output
+      torque is a sum of PD torque and additional torque.
+    """
 
     POSITION = 0
     TORQUE = 1
     HYBRID = 2
 
 
-# NOTE: Until functionality is added to MotorModel,
-# it is effectively equivalent to a `dataclass`.
+@dataclass
+class MotorCommand:
```

Comment:
This should have a short description of why it exists I think.

Reply:
Agreed, and thanks for the catch.

Does flake8 catch missing documentations? I know pylint allows that, but it requires extra configuration.

Reply:
You raise a good point.

I think now that we are getting a sense for what we want our documentation to look like, we can possibly start linting for that in CI too.

Here's one:
https://gitlab.com/pycqa/flake8-docstrings
also
https://github.com/peterjc/flake8-rst-docstrings (possibly additionally)

I can look into this more soon.